### ISSUE TITLE
[feat] `control-has-associated-label` can't be configured to support parent elements with label props

### ISSUE THREAD

**Issue by fnune** [jsx-eslint/eslint-plugin-jsx-a11y#987]
From https://github.com/jsx-eslint/eslint-plugin-jsx-a11y/blob/main/docs/rules/control-has-associated-label.md#rule-options

> labelAttributes is a list of attributes to check on the control component and its children for a label. Use this if you have a custom component that uses a string passed on a prop to render an HTML label, for example.

Unfortunately this only checks children elements of the component and won't allow for setups like this:

```tsx
<Form.Item label="...">
  <Input />
</Form.Item>
```

**Comment by ljharb** [comment]
At some complexity of abstractions there's not much static analysis can do for you. Is `<Input />` a placeholder here, or is it an actual component you're using?
